### ISSUE TITLE
Entity stays invisible when deleting Zone

### ISSUE THREAD

**Issue by JulianGro** [overte-org/overte#795]
When you delete the zone that an entity is using as part of "Render With Zones", the entity will stay hidden and display an "ERROR: NOT FOUND".
The error seems nice, but I would argue that an entity should not be culled if the zone that culls it does not exist.
![IMG_20240123_110519](https://github.com/overte-org/overte/assets/11144627/9e1918ed-ad6e-432e-8932-031e14e6f101)

**Comment by gydence** [comment]
yeah I agree that this is the correct behavior.  maybe when you delete a zone, its ID should be removed from all renderWithZones lists?  but we don’t currently keep track of that mapping anywhere and it would be a bit expensive to loop over all entities. we’d have to figure out where to put that logic…create or interface could not have loaded entities yet or might not have edit permission, so I guess it would be on the entity server

**Comment by AleziaKurdis** [comment]
There are other reasons to obtain a broken link by the way. (Copy of the entity in a different domain, import a json that has an id but with no zone )

Most of the time I prefer to have a clue that there was a broken link (in order to manage it)
If you delete a zone by accident you still have at least a clue of which entities were depending on it.

I would say: Caution with changing this behavior for something destructive.
It is probably not top priority to address anyway. Maybe we could wait a 10th user to complain, since we are like 3 people having use this feature. (yes I'm certainly exegerating... but not that much )

**Comment by gydence** [comment]
fair 😁

I think the entity list can show you when URLs are broken, right?  maybe it could also show a warning for this

**Comment by AleziaKurdis** [comment]
We could have this kind of thing yes.
it would be useful.
So more an enhancement anyway

**Comment by AleziaKurdis** [comment]
Maybe a tool specifically to manage the broken links.
It would list the id that are not resovled among the renderWithZones properties and offer the choice to: Remove or Replace them.
It would work better than the multiple selection. and control the list of the entities that you want to fix,

**Comment by AleziaKurdis** [comment]
I suggest we close this now.

**Comment by AleziaKurdis** [comment]
The find entity can’t be trusted for that, it might not not return entities that are far and small and never been visited in the current session. So such warning might fail to advise you and this entity will stay hidden anyway.

Also there is the case where you paste entities that come from another domain and have a value in renderWithZones. Same issue here. We can’t trust the Find entities, so zone zone might be wrongly considered as missing. So you might get the no warning and still get the pasted entities not visible.

**Comment by AleziaKurdis** [comment]
yeah... but create app should also work for serverless.
It sound like complicated for a usecase that seem questionable.
Can you explain what was the context when you experienced that?
Was that a test of a real case ?

**Comment by AleziaKurdis** [comment]
> separately, for the pasting issue, renderWithZones IDs _should_ get re-written when pasting entities so that they target the new ID. if that's not working as intended we should fix it

I undertsand that you will continue to set the id in renderwithzone only if the zone is part of the copy?
and of course discard it when it is not, right?

**Comment by HifiExperiments** [comment]
I think that's how it's supposed to work 😜
